File: jass-to-ts/src/main/java/org/vsx/jassToTs/JassTranslatorException.java
package org.vsx.jassToTs;

public class JassTranslatorException extends Exception {
    static boolean isStrict = true;
    public static boolean getIsStrict() {
        return isStrict;
    }
    public static void setIsStrict(boolean value) {
        isStrict = value;
    }
    public static void Error(String message) throws JassTranslatorException {
        if (isStrict) throw new JassTranslatorException(message);
        System.out.println(message);
    }
    public JassTranslatorException(String message) {
        super(message);
    }
}

File: JassToTs/JassToTs/JassTranslatorException.cs
using System;

namespace JassToTs {
    public class JassTranslatorException : Exception {
        static bool isStrict = true;
        public static bool IsStrict { get => isStrict; set => isStrict = value; }

        public static void Error(string message)
        {
            if (isStrict) throw new JassTranslatorException(message);
            Console.WriteLine(message);
        }
        public JassTranslatorException(string message): base(message) {}
    }
}